src/addressindex.h

```diff
@@ -215,4 +215,19 @@ struct CAddressIndexIteratorHeightKey {
     }
 };
 
+template <typename T1>
+inline std::vector<uint8_t> TrimScriptP2PK(const T1& input) {
```

Comment:
memory here should be owned indeed because:
 -  `uint160` has std::vector inside
 - this data is used outside of AddressBytesFromScript while original CScript should not be changed

Current solution is perfectly fine at the moment.